examples/step-52/doc/results.dox

```diff
@@ -10,28 +10,27 @@ The console output contains both errors and, for some of the methods, the
 number of steps they performed:
 @code
 Explicit methods:
-Forward Euler:            error=1.00883
-Third order Runge-Kutta:  error=0.000227982
-Fourth order Runge-Kutta: error=1.90541e-06
+   Forward Euler:            error=1.00883
+   Third order Runge-Kutta:  error=0.000227982
+   Fourth order Runge-Kutta: error=1.90541e-06
 
 Implicit methods:
-Backward Euler:           error=1.03428
-Implicit Midpoint:        error=0.00862702
-Crank-Nicolson:           error=0.00862675
-SDIRK:                    error=0.0042349
-
-Embedded %explicit methods:
-Heun-Euler:               error=0.0073012
-                steps performed=284
-Bogacki-Shampine:         error=0.000403281
-                steps performed=181
-Dopri:                    error=0.0165485
-                steps performed=119
-Fehlberg:                 error=0.00104926
-                steps performed=106
-Cash-Karp:                error=8.59366e-07
-                steps performed=107
+   Backward Euler:           error=1.03428
+   Implicit Midpoint:        error=0.00862702
+   Crank-Nicolson:           error=0.00862675
+   SDIRK:                    error=0.0042349
 
+Embedded explicit methods:
+   Heun-Euler:               error=0.0073012
+                   steps performed=284
+   Bogacki-Shampine:         error=0.000408407
+                   steps performed=181
+   Dopri:                    error=0.000836695
+                   steps performed=120
+   Fehlberg:                 error=0.00248922
+                   steps performed=106
+   Cash-Karp:                error=0.0787735
```

Comment:
why is this error so much bigger?

Reply:
In this case, my hypothesis is that the previous code was using a really small time-step at the end which caused the error to go way down. `DiscreteTime` objects can decide to extend the last time-step just a little bit to avoid a tiny final dt.
https://github.com/dealii/dealii/blob/ff89e12c96f2186d84e14dc3b7d798e2461710ce/source/base/discrete_time.cc#L27-L40

Unfortunately, I have not put print statements in either the old and the new code to confirm my hypothesis yet, but the hypothesis agrees with the following observations: 
* The error values don't change at all for most algorithms.
* The result for Cash-Karp has now one fewer time-steps and the error value which used to be unbelievably small (even below the coarsening threshold) is now reasonable (based on the adaptive time-stepping algorithm).

Reply:
Even before changing the code, the error values I got with my machine was different than those presented in the tutorial (similar to what is being discussed in #9684). What follows is the summary of discrepancies.
Results provided in the online tutorial:
```
Bogacki-Shampine:         error=0.000403281
                steps performed=181
Dopri:                    error=0.0165485
                steps performed=119
Fehlberg:                 error=0.00104926
                steps performed=106
Cash-Karp:                error=8.59366e-07
                steps performed=107
```
Results of the master branch on my machine (identical for both debug and release)
```
   Bogacki-Shampine:         error=0.000403281
                steps performed=181
   Dopri:                    error=5.07155e-05
                steps performed=121
   Fehlberg:                 error=0.00401035
                steps performed=106
   Cash-Karp:                error=0.00156445
                steps performed=107
```
Results after making my changes with `DiscreteTime` (identical both debug and release)
```
   Bogacki-Shampine:         error=0.000408407
                   steps performed=181
   Dopri:                    error=0.000836695
                   steps performed=120
   Fehlberg:                 error=0.00248922
                   steps performed=106
   Cash-Karp:                error=0.0787735
                   steps performed=106
```
This also supports the hypothesis that I mentioned in my previous comment. After making the changes, both Dopri and Cash-Karp get one fewer time step but higher level of error (still below the refinement threshold of 0.1).

Reply:
Yes, that's correct we already discussed this in https://github.com/dealii/dealii/issues/9684

Reply:
Sorry, I didn't follow that conversation very closely.

I guess the test problem is a bit unfortunate if the size of the final time step determines the total error of the solution.

Reply:
That's to be expected with the embedded methods https://github.com/dealii/dealii/issues/9684#issuecomment-627066689